compiler/src/iree/compiler/Dialect/HAL/Target/ROCM/ROCMTargetUtils.cpp

```diff
@@ -161,40 +162,41 @@ std::string createHsaco(const std::string isa, StringRef name) {
 
   // Invoke lld. Expect a true return value from lld.
   // Searching for LLD
-  std::string lldProgram;
-  std::string toolName = "ld.lld";
-  if (llvm::sys::fs::exists(toolName)) {
-    llvm::SmallString<256> absolutePath(toolName);
-    llvm::sys::fs::make_absolute(absolutePath);
-    lldProgram = std::string(absolutePath);
-  } else {
-    // Next search the environment path.
-    if (auto result = llvm::sys::Process::FindInEnvPath("PATH", toolName)) {
-      lldProgram = std::string(*result);
-    }
-  }
+  const SmallVector<std::string> &toolNames{"iree-lld"};
+  std::string lldProgram = findTool(toolNames);
   if (lldProgram.empty()) {
-    llvm::WithColor::error(llvm::errs(), name)
-        << "unable to find ld.lld in PATH\n";
+    llvm::WithColor::error(llvm::errs(), name) << "unable to find iree-lld.\n";
     return {};
   }
   // Setting Up LLD Args
+  if (lldProgram.front() == '"') {
+    lldProgram.erase(0, 1);                  // erase the first character
+    lldProgram.erase(lldProgram.size() - 1); // erase the last character
+  }
+#if defined(_WIN32)
+  llvm::StringRef lldName = "iree-lld.exe";
+#else
+  llvm::StringRef lldName = "iree-lld";
+#endif // _WIN32
   std::vector<llvm::StringRef> lldArgs{
```

Comment:
this dance should not be required - findTool should return exactly what you need and it should be directly usable (it handles .exe if needed, etc) - no need for string slicing and such

See EmbeddedLinkerTool.cpp for how to do this - you want precisely what it has

Reply:
I removed this as it should work as explained. Had a misconception about the inputs to llvm::sys::ExecuteAndWait()
I need to test that it's still functional -- if not, I'll fix the method of command execution to be exactly what we do in LLVMCPU as well.

Reply:
Switching to draft while I rework the utilities to match existing usage -- current version doesn't work